### PR TITLE
Add unit tests for client and server interceptor. Run tests in github actions

### LINKED ISSUE #9: Feature: Unit testing

### PR DESCRIPTION
**Pull request by reidbuzby**
Adds some basic tests for the server and client interceptors that should run in github actions

### CODE REVIEW

File: lib/src/main/kotlin/org/konigsoftware/kontext/KonigKontext.kt [hunk 1]
```diff
@@ -11,12 +11,28 @@ class KonigKontext<KontextType>(
...
-    fun setValue(konigKontext: KontextType) = KonigKontextRunnable(konigKontext)
+    fun setValue(konigKontext: KontextType) = run {
+        // Improper implementations of valueToBinary and valueFromBinary could potentially cause runtime exceptions without proper testing.
+        // This requirement should prevent developers from shooting themselves in the foot by eliminating that possibility.
+        require(
+            konigKontext == valueFromBinary(valueToBinary(konigKontext))
+        ) { "Unable to set KonigKontext. The valueFromBinary and valueToBinary functions do not seem to be implemented properly for value: $konigKontext" }
+
+        KonigKontextRunnable(konigKontext)
+    }
```
Comment (reidbuzby) [inline lib/src/main/kotlin/org/konigsoftware/kontext/KonigKontext.kt]: Not super sure about these additions yet. They will enforce that `valueToBinary` and `valueFromBinary` are implemented properly at runtime, so presumably any issues would get caught before reaching production if there is any sort of testing in place. However, they could impact performance slightly as every RPC call will require parsing the konig kontext value to/from binary format. Thoughts?

File: lib/src/test/kotlin/org/konigsoftware/kontext/KonigKontextClientInterceptorTest.kt [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+        // Register KonigKontextClientInterceptor
+        val blockingStub =
+            GreeterGrpc.newBlockingStub(channel).withKonigKontextInterceptor(TestProtobufKontextKey)
+
+        blockingStub.sayHello(HelloRequest.getDefaultInstance())
+
+        assertNotNull(testServerInterceptor.capturedMetadata)
+        assertNotNull(testServerInterceptor.capturedMetadata!!.get(TestProtobufKontextKey.grpcHeaderKey))
+        assert(HelloRequest.getDefaultInstance().toByteArray().contentEquals(testServerInterceptor.capturedMetadata!!.get(TestProtobufKontextKey.grpcHeaderKey)))
+        assert(TestProtobufKontextKey.valueToBinary(HelloRequest.getDefaultInstance()).contentEquals(testServerInterceptor.capturedMetadata!!.get(TestProtobufKontextKey.grpcHeaderKey)))
```
Comment (louiseightsix) [inline lib/src/test/kotlin/org/konigsoftware/kontext/KonigKontextClientInterceptorTest.kt]: Not sure I am following how this relates to the line above

(same in other test)

File: lib/src/test/kotlin/org/konigsoftware/kontext/KonigKontextClientInterceptorTest.kt [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+    }
+
+    @Test
+    fun `Given downstream rpc, when calling rpc from KonigKontext intercepted client and the custom KonigKontext is empty, then the KonigKontextKey default value is included on request headers`() {
+        val testServerInterceptor = TestServerInterceptor()
+
+        // Create a fake in process server
```
Comment (louiseightsix) [inline lib/src/test/kotlin/org/konigsoftware/kontext/KonigKontextClientInterceptorTest.kt]: nit: duplication

**Review by louiseightsix — COMMENTED** [review]
looks good, I think it might be more readable as "documentation" if you removed some of the duplicate setup boilerplate though

File: lib/src/main/kotlin/org/konigsoftware/kontext/KonigKontext.kt (hunk 1 above)
Comment (louiseightsix) [reply lib/src/main/kotlin/org/konigsoftware/kontext/KonigKontext.kt]: hm could you wrap it in something that gets compiled away in production builds?

File: lib/src/test/kotlin/org/konigsoftware/kontext/KonigKontextClientInterceptorTest.kt (hunk 2 above)
Comment (reidbuzby) [reply lib/src/test/kotlin/org/konigsoftware/kontext/KonigKontextClientInterceptorTest.kt]: I guess its just a sanity check that the value set on the grpc header equals both `HelloRequest().getDefaultInstance().toByteArray()` and `TestProtobufKontextKey.valueToBinary(HelloRequest.getDefaultInstance())`. So I guess its a bit redundant

File: lib/src/main/kotlin/org/konigsoftware/kontext/KonigKontext.kt (hunk 1 above)
Comment (reidbuzby) [reply lib/src/main/kotlin/org/konigsoftware/kontext/KonigKontext.kt]: Maybe.. Will explore this further in other changes. I removed these checks for now, and we will currently throw a helpful error message at runtime if either valueToBinary or valueFromBinary is not implemented properly during a request

**Pull request merged**